### PR TITLE
test: add unit tests

### PR DESCRIPTION
**Pull request by anik31**
### Files to check
- src/reducer/address-reducer.test.js
- src/reducer/cart-reducer.test.js
- src/reducer/product-reducer.test.js
- src/reducer/wishlist-reducer.test.js
- src/utils/discountCalc.test.js
- src/utils/filterData.test.js
- src/utils/searchData.test.js
- src/utils/sortData.test.js

image

### CODE REVIEW

File: src/reducer/address-reducer.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const expectedResult = {
+        _id: "27d94942-cd9b-4137-a0bd-39296b8a6bd2",
+        name: "Aniket Prakash",
+        street: "Near MKC High School",
+        city: "Keonjhar",
+        state: "Odisha",
+        pincode: "411001"
+      };
+
+    expect(state).toContainEqual(expectedResult);
```
Comment (Maianki) [inline src/reducer/address-reducer.test.js:29]: query: Any particular reason you have used toContainEqual instead of toEqual here?

**Review by Maianki — APPROVED** [review]
Great job Aniket. Everything looks good. 
Suggestion : You can setup github action.

File: src/reducer/address-reducer.test.js (hunk 1 above)
Comment (anik31) [reply src/reducer/address-reducer.test.js:29]: The state is an array of objects whereas expectedResult is an object.

**Review by sainath7878 — APPROVED** [review]
Great job on adding test cases. Good to merge

File: src/reducer/address-reducer.test.js (hunk 1 above)
Comment (Maianki) [reply src/reducer/address-reducer.test.js:29]: okay, thanks.

**Comment by anik31** [comment]
Thanks, @Maianki & @sainath7878 for reviewing, I will add GitHub actions later.

**Pull request merged**